src/utils/Api.js

```diff
@@ -0,0 +1,115 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getUserInfo(), this.getInitialCards()]);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it, you need to make a special `_checkResponse` method that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call.